doc/hma-ui.md

```diff
@@ -0,0 +1,280 @@
+* Each contains 5,000 sample hashes  
+* Try uploading images to test matching for content in hash banks
+
+
+### Creating a Demo Bank
+
+Users can create a hash bank separate from external hash exchanges like the NCMEC hash list or GIFCT hash list. Simply click “Create New Bank” and manually upload a few files to test.
+
+Once you add content, you can test by uploading either the same or different image in the “Find Content in Banks” section. If there is a match, the UI will show which hash bank the content was found in.
+
+
+### Browsing Hash Banks
+
+The web UI only allows searching for specific files within the hash banks. If users want to explore an overview or browse content in the hash banks, they have to query the database directly.
+
+#### **Database Table Stats (Quickest Overview)**
+
+Run this command to see high-level statistics:
+
+```shell
+flask --app OpenMediaMatch.app table_stats
+```
+
+This shows counts of banks, contents, signals, etc.
+
+#### **Direct Database Queries**
+
+You can also connect to PostgreSQL directly to browse the data:
+
+```shell
+# Connect to database  
+psql -h localhost -p 5432 -U media_match -d media_match
+# List all banks 
+SELECT * FROM bank;
+# See all content in a specific bank 
+SELECT bc.id, bc.bank_id, b.name, cs.signal_type, cs.signal_val 
+FROM bank_content bc 
+JOIN bank b ON bc.bank_id = b.id 
+JOIN content_signal cs ON bc.id = cs.content_id 
+WHERE b.name = 'SEED_BANK_0';
+# Count contents per bank 
+SELECT b.name, COUNT(bc.id) as content_count 
+FROM bank b 
+LEFT JOIN bank_content bc ON b.id = bc.bank_id 
+GROUP BY b.name;
+```
+
+#### **REST API Endpoints**
+
+The API provides several endpoints to browse banks:  
+**List all banks:**
+
+```shell
+curl http://localhost:5000/c/banks
+```
+
+**Get specific bank info:**
+
+```shell
+curl http://localhost:5000/c/bank/SEED_BANK_0
+```
+
+**Get specific content details:**
+
+```shell
+curl http://localhost:5000/c/bank/SEED_BANK_0/content/1
+```
+
+## **Exchanges (`/ui/exchanges`)**
+
+The Exchanges page lets users configure external hash exchanges that are API based.
+
+
+### Supported Hash Exchanges
+
+**NCMEC (National Center for Missing & Exploited Children)**
+
+* Child safety hash sharing program  
+* Requires authentication credentials  
+* Automated hash synchronization
+
+**GIFCT (Global Internet Forum to Counter Terrorism)**
+
+* Terrorist content hash database  
+* Enterprise-level access required  
+* Real-time threat intelligence updates
+
+**Meta ThreatExchange**
+
+* Comprehensive threat intelligence platform  
+* API key authentication required  
+* Multi-signal type support
+
+**StopNCII**
+
+* Non-consensual intimate image hashes  
+* Privacy-focused hash sharing  
+* Victim-centered approach
+
+### Exchange Configuration
+
+Several of the hash exchanges supported require credentials, provided in the form of a JSON blob:
+
+1. Configure authentication credentials (API keys, certificates)  
```

Comment:
We might need to give a hint on the authentication configuration. This might also be a documentation need for us on the exchanges themselves - we might want to include a future README.md in https://github.com/facebook/ThreatExchange/tree/main/python-threatexchange/threatexchange/exchanges which includes the collaboration config explainer and which credentials are accepted, which hopefully is just copying docs from the code.